Grbl_Esp32/src/NutsBolts.h

```diff
@@ -72,8 +72,11 @@
 #define isequal_position_vector(a, b) !(memcmp(a, b, sizeof(float) * N_AXIS))
 
 // Bit field and masking macros
-//Arduino.h:104:0: note: this is the location of the previous definition
-//#define bit(n) (1 << n)
+// bit(n) is defined in Arduino.h.  We redefine it here so we can apply
+// the static_cast, thus making it work with scoped enums
+#undef bit
+#define bit(n) (1 << static_cast<unsigned int>(n))
+
```

Comment:
Hmm I always find this kinds of things scary because now developers don't know which define they use. However, I do see this as a temporary solution, and if compatible with the code in Arduino there ain't no problem.

So about that... I just checked to be sure. Arduino defines `#define bit(n) (1 << n)`. 1 is an int, n here is an arbitrary type. The return type depends on `1`, which is an `int`, hence the return value is an `int`. Same for the solution you have here. In other words, it is correct.

You could also get rid of the define and make it type-safe, by using a constexpr function. The behavior is that it will generate a constant if a constant is passed, or a value if a non-constant variable is passed:

`template <typename T> constexpr int bit(T n) { return 1 << static_cast<unsigned int>(n);}`

Reply:
Let's figure out the right way to handle this in a later phase, where we sort out what to do about the bucketload of bitfield types in System.h